src/API/FeedState.php

```diff
@@ -163,26 +164,20 @@ public function get_feed_state() {
 	 */
 	private function add_local_feed_state( $result ) {
 
-		$local_feed = Pinterest\ProductFeedStatus::get_local_feed();
 		$state      = Pinterest\ProductFeedStatus::get();
 		$extra_info = '';
 
 		switch ( $state['status'] ) {
-			case 'starting':
-				$status       = 'pending';
-				$status_label = esc_html__( 'Feed generation is being initialized.', 'pinterest-for-woocommerce' );
-				break;
 
 			case 'in_progress':
 				$status       = 'pending';
 				$status_label = esc_html__( 'Feed generation in progress.', 'pinterest-for-woocommerce' );
 				$extra_info   = sprintf(
-					/* Translators: %1$s Time string, %2$s current index of written products, %3$s total number of products, %4$s Opening link tag, %5$s closing link tag */
-					esc_html__( 'Last activity: %1$s ago - Wrote %2$s out of %3$s products to %4$sfeed file%5$s.', 'pinterest-for-woocommerce' ),
+					/* Translators: %1$s Time string, %2$s number of products */
+					esc_html__( 'Last activity: %1$s ago - Wrote %2$s products to %3$sfeed file%4$s.', 'pinterest-for-woocommerce' ),
 					human_time_diff( $state['last_activity'] ),
-					$state['current_products'],
 					$state['product_count'],
-					'<a href="' . $local_feed['feed_url'] . '" target="_blank">',
+					'<a href="' . $this->get_feed_url() . '" target="_blank">',
```

Comment:
For readability, I think `sprintf()` would be preferable here, and we should also be escaping the URL.

```suggestion
					sprintf( '<a href="%s" target="_blank">', esc_url( $this->get_feed_url() ) ),
```

Also, do we want to have any kind of handling (either here or in the `get_feed_url()` method) for getting no result back for the local feed?

Reply:
sprintf is a good suggestion.
If the feed was not generated yet, this part of UI is not used, so we don't need to worry about no results.